src/Controller/ModulesController.php

```diff
@@ -350,6 +350,26 @@ protected function saveRelated(array $relatedData, string $id): void
             $method = (string)Hash::get($rel, 'method');
             $relation = (string)Hash::get($rel, 'relation');
             $relatedIds = (array)Hash::get($rel, 'relatedIds');
+
+            foreach ($relatedIds as $key => &$data) {
+                if (!empty($data['attributes'])) {
+                    // avoid empty objects
+                    $saveObject = false;
+                    foreach ($data['attributes'] as $key => $field) {
```

Comment:
Using the same `$key` variable in cascade foreach:
```
foreach ($relatedIds as $key => &$data) { // line 354
[...]
    foreach ($data['attributes'] as $key => $field) { // line 358
```
Better using another variable name (`$k` or something else).

Reply:
fixed